frameworks/AutoGluonTS/setup.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
```

Comment:
it is the same setup as for default `AutoGluon`, right?
Why create another framework then? It adds a lot of complexity regarding testing, distribution, docker images and so on...
In the `AutoGluon.__init__.py` you could just fork the logic like this:
```python
exec_file = "exec_ts.py" if dataset.type is not DatasetType.timeseries else "exec.py"
return run_in_venv(__file__, exec_file,
                       input_data=data, dataset=dataset, config=config)
```

Reply:
One difference @limpbot, I install MXNet in addition to the other normal dependencies for TimeSeries, since it isn't a default install. We can simply install MXNet by default for now, hopefully it won't cause issues.

@sebhrusen The one concern is if AutoGluon becomes too monolithic of an install, we may want to consider having separate install logic for the different submodules that are unrelated to eachother (for example, timeseries doesn't need vision and text modules, tabular doesn't need timeseries module). Probably not needed now, but something to keep in mind since AutoGluon covers more data types/domains than most AutoML systems and that comes with many dependencies.

Reply:
@Innixma I understand the concern.
for better encapsulation and to allow you to reuse code easily in benchmarks when using different submodules, I'd still advise to keep a single `AutoGluon` folder.
You can then provide different flavors for the setup just in the framework definition using the `setup_env` syntax:
```yaml
AutoGluon_TimeSeries
  extends: AutoGluon
  setup_env:
    MODULE: timeseries
    VAR: string
```
this makes the 2 variables `MODULE` and `VAR` directly available in `setup.sh` (right after the call to `. ${HERE}/../shared/setup.sh ${HERE} true`) and allows you to customize the setup: you may already be using this for dev environment.

Also, thanks to the definitions hierarchy (`extends: AutoGluon`), maybe we can later tweak the results to make it appear as just `AutoGluon`, or we can add a notion of group, whatever...


This may not be perfect when you switch frequently between definitions, but for now, I'd like to keep the framework folders to a minimum.
I agree with you that we probably need to start thinking about distinguishing the setup/exec for different kind of tasks, ideally it should be smooth and not even require the additional definition above: for a given type of task, the framework should be able to tell early if it can handle it, if not it could try to apply some additional setup before replying, and if it's ready handle it, then it continues as before: but all of this is much easier to change if there's already one single folder in the first place.
I can create an issue for this, although I don't have huge time to dedicate to AMLB lately, but this may change in a couple of months.